pkg/adapter/kubernetesevents/adapter.go

```diff
@@ -101,11 +99,16 @@ func (a *Adapter) addEvent(new interface{}) {
 	logger := logging.FromContext(context.TODO()).With(zap.Any("eventID", event.ObjectMeta.UID), zap.Any("sink", a.SinkURI))
 	logger.Debug("GOT EVENT", event)
 
-	if a.client == nil {
-		a.client = cloudevents.NewClient(a.SinkURI, cloudevents.Builder{EventType: eventType})
+	if a.ceClient == nil {
+		var err error
+		a.ceClient, err = client.NewHttpClient(context.TODO(), a.SinkURI, http.BinaryV02)
```

Comment:
Can any of the arguments default?

context seems like it should be `context.Background()`, since you're not forwarding the context from elsewhere, you're acting on your own authority.

Reply:
HECK YEAH NOW THEY CAN!!!